module/zfs/zfs_fm.c

```diff
@@ -131,7 +131,8 @@ zfs_ereport_start(nvlist_t **ereport_out, nvlist_t **detector_out,
 	 * If we are doing a spa_tryimport() or in recovery mode,
 	 * ignore errors.
 	 */
-	if (spa_load_state(spa) == SPA_LOAD_TRYIMPORT ||
+	if (spa != NULL ||
+	    spa_load_state(spa) == SPA_LOAD_TRYIMPORT ||
```

Comment:
Right the fix itself is wrong here.  It should be `spa == NULL` in the check so it returns immediately return when no spa is passed.  Or better yet just drop think hunk completely.  This is an internal function all callers know to always pass a spa.  Period.